packages/mobile/src/screens/explore-screen/components/ActiveDiscussions.tsx

```diff
@@ -4,27 +4,26 @@ import { useRecentlyCommentedTracks } from '@audius/common/api'
 import { exploreMessages as messages } from '@audius/common/messages'
 import { QueueSource } from '@audius/common/store'
 
+import { useDeferredElement } from '../../../hooks/useDeferredElement'
+
 import { ExploreSection } from './ExploreSection'
 import { TrackTileCarousel } from './TrackTileCarousel'
 
 export const ActiveDiscussions = () => {
-  const { data: recentlyCommentedTracks, isLoading } =
-    useRecentlyCommentedTracks({ pageSize: 10 })
-
-  if (
-    !isLoading &&
-    (!recentlyCommentedTracks || recentlyCommentedTracks.length === 0)
-  ) {
-    return null
-  }
+  const { inView, InViewWrapper } = useDeferredElement()
+  const { data: recentlyCommentedTracks, isPending } =
+    useRecentlyCommentedTracks({ pageSize: 10 }, { enabled: inView })
 
   return (
-    <ExploreSection title={messages.activeDiscussions}>
-      <TrackTileCarousel
-        tracks={recentlyCommentedTracks}
-        isLoading={isLoading}
-        source={QueueSource.EXPLORE}
-      />
-    </ExploreSection>
+    <InViewWrapper>
+      <ExploreSection title={messages.activeDiscussions}>
+        <TrackTileCarousel
+          tracks={recentlyCommentedTracks}
+          isLoading={isPending}
+          source={QueueSource.EXPLORE}
+          inView={inView}
```

Comment:
a little bit of abstraction bleed here. Can we just do `isLoading={isPending || !inView}`? Would be nice for the layout component (carousel) to not be aware of this optional deferred loading pattern and keep that isolated just to the section components themselves.

Reply:
yeah completely agree